mod/discussions/views/default/widgets/discussions/content.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * User blog widget display view
+ */
+
+use Elgg\Database\Clauses\OrderByClause;
+
+/* @var $widget \ElggWidget */
+$widget = elgg_extract('entity', $vars);
+
+$num_display = (int) $widget->num_display ?: 4;
+
+$options = [
+	'type' => 'object',
+	'subtype' => 'discussion',
+	'limit' => $num_display,
+	'pagination' => false,
+	'distinct' => false,
+	'order_by' => new OrderByClause('e.last_action', 'desc'),
+];
+
+$owner = $widget->getOwnerEntity();
+$url = elgg_generate_url('collection:object:discussion:all');
+
+if ($owner instanceof \ElggUser) {
+	if ($widget->context === 'dashboard') {
+		if (elgg_get_plugin_setting('enable_global_discussions', 'discussions') !== 1) {
+			$groups = $owner->getGroups([
+				'limit' => false,
+				'batch' => true,
+			]);
+			
+			$group_guids = [];
+			foreach ($groups as $group) {
+				$group_guids[] = $group->guid;
+			}
+			
+			if (!empty($group_guids)) {
+				$options['container_guids'] = $group_guids;
+				$url = elgg_generate_url('collection:object:discussion:my_groups', ['username' => $owner->username]);
+			}
+		}
+	} else {
+		$options['owner_guid'] = $owner->guid;
+		$url = elgg_generate_url('collection:object:discussion:owner', ['username' => $owner->username]);
+	}
+} elseif ($owner instanceof \ElggGroup) {
+	$options['container_guid'] = $owner->owner_guid;
+	$url = elgg_generate_url('collection:object:discussion:group', ['guid' => $owner->guid]);
+}
+
+$content = elgg_list_entities($options);
+if (empty($content)) {
+	echo elgg_echo('discussion:none');
```

Comment:
use `page/components/no_results`

Reply:
keeping this inline with other content plugins... reasons is probably excessive whitespace issues if using the page component